### PR TITLE
[WIP] Update testcase for cert plugin

### PR DESCRIPTION
**Pull request by Akasurde**
Fixes https://fedorahosted.org/freeipa/ticket/6275

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

### CODE REVIEW

File: ipatests/test_xmlrpc/test_cert_plugin.py [hunk 1]
```diff
@@ -191,10 +192,10 @@ def test_0005_cert_uris(self):
 
         See https://fedorahosted.org/freeipa/ticket/5881
         """
-        result = api.Command.cert_show(sn, out=unicode(self.certfile))
+        api.Command['cert_show'](sn, out=unicode(self.certfile))
```
Comment (MartinBasti) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: Why this change?

File: ipatests/test_xmlrpc/test_cert_plugin.py [hunk 2]
```diff
@@ -499,3 +506,34 @@ def test_revoke_with_reason_9(self):
...
+
+        assert 'cert_request_status' in res
+        assert 'complete' in res['cert_request_status']
+
+    def test_0002_cert_status_ca(self):
+        """
+        Show status of certificate using request id and CA
+        """
+        res = api.Command['cert_status'](request_id=1,
+                                         cacn=unicode('ipa'))['result']
```
Comment (MartinBasti) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: please use `u'ipa'` instead of `unicode(ipa)`

File: ipatests/test_xmlrpc/test_cert_plugin.py [hunk 3]
```diff
@@ -499,3 +506,34 @@ def test_revoke_with_reason_9(self):
...
+
+    """
+    Test the `cert-status` command.
+    """
+
+    def test_0001_cert_status(self):
+        """
+        Show status of certificate using request id
+        """
+        res = api.Command['cert_status'](request_id=1)['result']
```
Comment (MartinBasti) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: I'm not sure, can we always assume that request with id=1 is there?

File: ipatests/test_xmlrpc/test_cert_plugin.py [hunk 4]
```diff
@@ -97,15 +100,14 @@ def run_certutil(self, args, stdin=None):
...
+        self.reqdir = tempfile.mkdtemp(prefix="tmp-")
         self.reqfile = self.reqdir + "/test.csr"
         self.pwname = self.reqdir + "/pwd"
         self.certfile = self.reqdir + "/cert.crt"
 
         # Create an empty password file
-        fp = open(self.pwname, "w")
-        fp.write("\n")
-        fp.close()
+        with open(self.pwname, "w") as fp:
```
Comment (MartinBasti) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: This should be in separate commit

File: ipatests/test_xmlrpc/test_cert_plugin.py [hunk 5]
```diff
@@ -122,9 +124,8 @@ def generateCSR(self, subject):
                            "-f", self.pwname,
                            "-a",
                            ])
-        fp = open(self.reqfile, "r")
-        data = fp.read()
-        fp.close()
```
Comment (MartinBasti) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: This refactoring should be in separate commit

File: ipatests/test_xmlrpc/test_cert_plugin.py [hunk 6]
```diff
@@ -50,6 +50,7 @@
 _EXP_CRL_URI = ''.join(['http://ipa-ca.', _DOMAIN, '/ipa/crl/MasterCRL.bin'])
 _EXP_OCSP_URI = ''.join(['http://ipa-ca.', _DOMAIN, '/ca/ocsp'])
 
+
```
Comment (MartinBasti) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: This PEP8fix should be in separate commit

File: ipatests/test_xmlrpc/test_cert_plugin.py [hunk 7]
```diff
@@ -97,15 +100,14 @@ def run_certutil(self, args, stdin=None):
         return ipautil.run(new_args, stdin)
 
     def setup(self):
-        self.reqdir = tempfile.mkdtemp(prefix = "tmp-")
+        self.reqdir = tempfile.mkdtemp(prefix="tmp-")
```
Comment (MartinBasti) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: Pep8 fix should be in separate commit

**Comment by MartinBasti** [comment]
I left some inline comments, this improves the test but it still misses several features to be tested.
You can finish these improvements and it can be pushed and add more improvements in a new PR

File: ipatests/test_xmlrpc/test_cert_plugin.py (hunk 6 above)
Comment (Akasurde) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: OK. I will keep this in mind for future.

File: ipatests/test_xmlrpc/test_cert_plugin.py (hunk 7 above)
Comment (Akasurde) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: OK.

File: ipatests/test_xmlrpc/test_cert_plugin.py (hunk 4 above)
Comment (Akasurde) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: OK.

File: ipatests/test_xmlrpc/test_cert_plugin.py (hunk 1 above)
Comment (Akasurde) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: result is unused and wanted to make api call uniform with other calls.

File: ipatests/test_xmlrpc/test_cert_plugin.py (hunk 3 above)
Comment (Akasurde) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: We will surely have `request_id=1` which is used for IPA CA certificate. That's why I used request id 1. We can change this if this is not the case.

File: ipatests/test_xmlrpc/test_cert_plugin.py (hunk 2 above)
Comment (Akasurde) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: OK.

**Comment by Akasurde** [comment]
@MartinBasti I working on other improvements and will update this PR accordingly.

**Comment by MartinBasti** [comment]
@Akasurde what is your opinion about creating a Tracker class for certificate?

**Comment by Akasurde** [comment]
@MartinBasti Will implement tracker class in different PR.

File: ipatests/test_xmlrpc/test_cert_plugin.py (hunk 1 above)
Comment (MartinBasti) [inline ipatests/test_xmlrpc/test_cert_plugin.py]: ok

**Comment by tomaskrizek** [comment]
Closing for inactivity. If this PR is still relevant, please re-open it.
